pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -74,7 +75,20 @@ def on_trainer_init(
                 " in v1.7. Please pass `pytorch_lightning.callbacks.progress.ProgressBar` with"
                 " `process_position` directly to the Trainer's `callbacks` argument instead."
             )
-        self.trainer._progress_bar_callback = self.configure_progress_bar(progress_bar_refresh_rate, process_position)
+
+        if progress_bar_refresh_rate is not None:
+            rank_zero_deprecation(
+                f"Setting `Trainer(progress_bar_refresh_rate={progress_bar_refresh_rate})` is deprecated in v1.5 and"
+                " will be removed in v1.7. Please pass `pytorch_lightning.callbacks.progress.ProgressBar` with"
+                " `refresh_rate` directly to the Trainer's `callbacks` argument instead."
+            )
+
+        if enable_progress_bar:
+            self.trainer._progress_bar_callback = self.configure_progress_bar(
+                progress_bar_refresh_rate, process_position
+            )
```

Comment:
what happens if progress_bar_refresh_rate=0 and enable_progress_bar=True? 
we still have to honor backwards compat for users setting progress_bar_refresh_rate=0 and expecting the progress bar to be disabled until v1.7

Reply:
> what happens if progress_bar_refresh_rate=0 and enable_progress_bar=True?
same as before, we don't create a progress bar. the logic in configure_progress_bar isn't changing